lib/eth/client.rb

```diff
@@ -121,8 +121,9 @@ def transfer_and_wait(destination, amount, sender_key = nil, legacy = false)
     # @param amount [Integer] the transfer amount in Wei.
     # @param sender_key [Eth::Key] the sender private key.
     # @param legacy [Boolean] enables legacy transactions (pre-EIP-1559).
+    # @param **nonce [Integer] optional specific nonce for transaction.
     # @return [String] the local transaction hash.
-    def transfer(destination, amount, sender_key = nil, legacy = false)
+    def transfer(destination, amount, sender_key = nil, legacy = false, **kwargs)
```

Comment:
Do you mind tow rewrite the sender key and legacy transaction toggle to kwargs too? I meant to do this for a while. 🙏🏼

Reply:
I thought about it, but it will crash backward compatibility.

Ok, I will do next steps for this PR:
- move some args to kwargs
- code quality
- conflicts resolving

Reply:
Done with https://github.com/q9f/eth.rb/pull/169/commits/ef5a07e24ab1d0795fab28d3c042674e33e7a1d1